mysite/announcements/models.py

```diff
@@ -32,3 +31,27 @@ def clean(self):
             raise ValidationError("The date and time can't be in past")
         if self.date_time_expire <= self.date_time_to_publish:
             raise ValidationError("Scheduled time can't be less than expiry time")
+
+
+@python_2_unicode_compatible
+class AnnouncementDeliveryStatus(models.Model):
+    SEEN = 0
+    YET_TO_RECEIVE = 1
+    EXPIRED = 2
+    USER_LEFT_GROUP = 3
+
+    STATUS_CHOICES = (
+        (SEEN, 'SEEN'),
+        (YET_TO_RECEIVE, 'YET TO RECEIVE'),
+        (EXPIRED, 'EXPIRED'),
+        (USER_LEFT_GROUP, 'USER LEFT GROUP')
+    )
+    announcement = models.ForeignKey(Announcements, on_delete=models.CASCADE)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    status_last_update_time = models.DateTimeField('status last update time',
+                                                   help_text='The time at status got updated')
+    status = models.IntegerField(choices=STATUS_CHOICES)
```

Comment:
Let's make this SmallPositiveIntegerField instead.

WHy?
- Save more space.
- Less space means more money savings.